tensorflow/contrib/distribute/python/cross_tower_utils.py

```diff
@@ -21,7 +21,11 @@
 import collections as pycoll
 import threading
 
-from tensorflow.contrib import nccl
+from tensorflow.python import pywrap_tensorflow
+if pywrap_tensorflow.IsGoogleCudaEnabled():
+  from tensorflow.contrib import nccl
+else:
+  from tensorflow.contrib import rccl as nccl
```

Comment:
should this be checked with `IsBuiltWithROCm`?

Reply:
please refer to `tensorflow/core/util/port.h`

Reply:
This is the exact same code stub from tf.contrib.all_reduce, so if this needs to change it should be updated there, as well.  I recall asking how to do this check on slack but I can't find the discussion now.

Reply:
@jeffdaily yeah I think it's worthwhile to change both places for the logic be upstreamed. please also check `tensorflow/python/framework/test_util.py` you can see such python API has been added.

Reply:
Would it be appropriate to piggyback on this PR to update both places?

Reply:
sure thing. please do so.